### PR TITLE
feat: Send buffer type to cspell

### PR DESCRIPTION
**Pull request by Sloff**
In order to include language specific words as well

### CODE REVIEW

**Comment by Sloff** [comment]
For example in Rust, before this change:
image

After change:
image

File: lua/null-ls/builtins/diagnostics/cspell.lua [hunk 1]
```diff
@@ -8,7 +8,13 @@ return h.make_builtin({
     filetypes = {},
     generator_opts = {
         command = "cspell",
-        args = { "stdin" },
+        args = function(params)
+            return {
+                "--language-id",
+                vim.api.nvim_buf_get_option(params.bufnr, "filetype"),
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/cspell.lua:14]:
```suggestion
                params.ft,
```

A small optimization. Looks good to me otherwise!

Comment (Sloff) [reply lua/null-ls/builtins/diagnostics/cspell.lua:14]: Awesome, thanks @jose-elias-alvarez!

**Comment by jose-elias-alvarez** [comment]
Awesome, thanks!

**Pull request merged**